roles/openshift_deploy_registry/tasks/main.yml

```diff
@@ -0,0 +1,24 @@
+---
+- name: Deploy registry
+  shell: osadm registry --create --credentials=/var/lib/openshift/openshift.local.certificates/openshift-registry/.kubeconfig --images='registry.access.redhat.com/openshift3_beta/ose-${component}:${version}'
+
+- name: Get registry deployment config
+  shell: osc get deploymentConfig/docker-registry -o json | sed -e '/"resourceVersion"/d'
+  register: output
+
+- name: Set registry node region
+  set_fact:
+    registry_fact: "{{ output.stdout | from_json | oo_set_resource_node({{ocom_infra_node_label}}) }}"
+
+- name: Write registry deployment config to temp file
+  copy:
+    content: "{{ registry_fact }}"
+    dest: /tmp/registry.json
+
+- name: Import new registry deployment config
+  shell: osc update deploymentConfig/docker-registry -f /tmp/registry.json
+
+- name: Remove registry deployment config temp file
+  file:
+    path: /tmp/registry.json
+    state: absent
```

Comment:
I agree, the problem is the image streams definition file is a giant json blob https://github.com/openshift/training/blob/master/beta3/image-streams.json
Maybe if we had an easier way to import the image streams but as it stands I really dislike the idea of storing a json blob in a var and expect people to override it with what ever they choose.